modules/civiremote_event/src/Form/RegisterForm.php

```diff
@@ -171,7 +171,10 @@ public static function defaultValue($field, $field_name, $type) {
       case 'checkboxes':
         if (
           isset($field['value'])
-          && array_key_exists($field['value'], $field['options'])
+          && (
+            is_array($field['value']) && [] === array_diff($field['value'], array_keys($field['options']))
+            || (is_int($field['value']) || is_string($field['value'])) && array_key_exists($field['value'], $field['options'])
+          )
         ) {
           $default_value = $field['value'];
```

Comment:
https://api.drupal.org/api/drupal/core%21lib%21Drupal%21Core%21Render%21Element%21Checkboxes.php/function/Checkboxes%3A%3AprocessCheckboxes/10 looks to me like `#default_value` always has to be an array (or `0`) because it is used in a call of `in_array()`. This would mean that this line should be:
```php
$default_value = (array) $field['value'];
```

Though I've not completely checked how this value is used...

Reply:
I guess you're right. Also, the Drupal Form API should take care of disallowed values and showing an "invalid selection" error.